libbindgen/src/ir/item.rs

```diff
@@ -205,7 +205,12 @@ impl TypeCollector for Item {
             ItemKind::Var(ref var) => {
                 types.insert(var.ty());
             }
-            _ => {} // FIXME.
+            ItemKind::Module(_) => {
+                // Module -> children edges are "weak", and we do not want to
+                // trace them. If we did, then whitelisting wouldn't work as
+                // expected: everything in every module would end up
+                // whitelisted.
```

Comment:
Its because we trace the `parent_id` to find the containing module, which will then end up tracing everything in the module, and the module's parent module which will end up tracing everything in that module, etc etc.